src/Error/Debug/ConsoleFormatter.php

```diff
@@ -0,0 +1,219 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.1.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Error\Debug;
+
+use RuntimeException;
+
+/**
+ * A Debugger formatter for generating output with ANSI escape codes
+ */
+class ConsoleFormatter implements FormatterInterface
+{
+    /**
+     * text colors used in colored output.
+     *
+     * @var array
+     */
+    protected $styles = [
+        // bold yellow
+        'const' => '1;33',
+        // green
+        'string' => '0;32',
+        // bold blue
+        'number' => '1;34',
+        // cyan
+        'class' => '0;36',
+        // grey
+        'punct' => '0;8',
+        // black
+        'property' => '0;30',
+        // magenta
+        'visibility' => '0;35',
+        // red
+        'special' => '0;31',
+    ];
+
+    /**
+     * Check if the current environment supports ANSI output.
+     *
+     * @return bool
+     */
+    public static function environmentMatches(): bool
+    {
+        if (PHP_SAPI !== 'cli') {
+            return false;
+        }
+        // NO_COLOR in environment means no color.
+        if (env('NO_COLOR')) {
+            return false;
+        }
+        // Windows environment checks
+        if (
+            DIRECTORY_SEPARATOR === '\\' &&
+            (!(bool)env('ANSICON') || env('ConEmuANSI') !== 'ON')
+        ) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Style text with ANSI escape codes.
+     *
+     * @param string $style The style name to use.
+     * @param string $text The text to style.
+     * @return string The styled output.
+     */
+    protected function style(string $style, string $text): string
+    {
+        $code = $this->styles[$style];
+
+        return "\033[{$code};m{$text}\033[0m";
```

Comment:
I'm not sure, but is that closing semicolon there actually valid?

```suggestion
        return "\033[{$code}m{$text}\033[0m";
```

Reply:
It seems to work for me in iTerm with and without the `;` so we can do without.